packages/wrangler/src/publish/index.ts

```diff
@@ -176,6 +176,12 @@ export function publishOptions(yargs: Argv<CommonYargsOptions>) {
 				describe:
 					"Send Trace Events from this worker to Workers Logpush.\nThis will not configure a corresponding Logpush job automatically.",
 			})
+			.option("experimental", {
```

Comment:
Why is this necessary? Why not just _try_ uploading and let the Workers API yell at users who try flags that aren't enabled on their account?

If we do decide to keep something in Wrangler for this, I'd personally consider moving control to an environment variable or adding `{ hidden: true }` so it's a bit more obscure. If we open it up to non-Cloudflare employees, then exposing it here makes sense; but until then, we should probably hide it?

Reply:
Also the reason we went with the cli option over an env var is that it is very clear that worker being published is not stable, whereas a var could be changed once to allow for the feature and the fact that the worker could break at any moment can be overlooked.

Reply:
`experimental` is already used as an object elsewhere in wrangler to enable experimental CLI features, and is named such to invite usage by early adopters. Naming an internal-only feature `experimental` will invite folks to try build on it. 

In the past we've named internal-only features: `logfwdr` and `first_party_worker` - may I suggest `experimental_first_party_worker` instead?

Reply:
I echo what others have said about the flag here. Users will quickly find a new flag and explore what it is and how to use it. Making it more obscure or explicit would be best.

Reply:
> I echo what others have said about the flag here. Users will quickly find a new flag and explore what it is and how to use it. Making it more obscure or explicit would be best.
> 
> I'm also not sure I fully understand the problem and proposed solution. We can talk internally about this, but a thought is also - how will we validate that the user utilizing this is an internal account? If we can do that validation, what's the point of this flag?

[The account validation is done in EWC](cflare.co/ewc-experimental-flag-gate), so no validation needs to be done in Wrangler, hence only the metadata Field being edited. The point of the flag is some insurance for the Runtime Team, we don't want someone creating a project and it then being taken over by someone who wouldn't know that experimental feature flags are being used. We really don't want people to rely on something that can/will break, but we do want people to play with the features (Just not make critical infrastructure for Cloudflare on them). [For all of the gory details on why this is being done](cflare.co/ewc-experimental-flag-gate-discussion)

**For non Cloudflare watchers:** The links above uses an internal link shortener that requires one to be part of Cloudflare to access it. This is so that internal infrastructure is not leaked.

Reply:
> `experimental` is already used as an object elsewhere in wrangler to enable experimental CLI features, and is named such to invite usage by early adopters. Naming an internal-only feature `experimental` will invite folks to try build on it.
> 
> In the past we've named internal-only features: `logfwdr` and `first_party_worker` - may I suggest `experimental_first_party_worker` instead?

That does make sense, it is made clear in the help text that only internal. Also [there are plans on deprecating this eventually](cflare.co/not-permanent-cli-flag), this is only a stop gap. Would `wrangler publish --enable-experimental-feature-flags` work?

**For non Cloudflare watchers:** The links above uses an internal link shortener that requires one to be part of Cloudflare to access it. This is so that internal infrastructure is not leaked.